app/src/main/java/by/homework/hlazarseni/tfgridexplorer/fragments/NodeDetailFragment.kt

```diff
@@ -0,0 +1,148 @@
+package by.homework.hlazarseni.tfgridexplorer.fragments
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.viewmodel.initializer
+import androidx.lifecycle.viewmodel.viewModelFactory
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import androidx.navigation.ui.setupWithNavController
+import by.homework.hlazarseni.tfgridexplorer.R
+import by.homework.hlazarseni.tfgridexplorer.constants.DbConstants.CONVERTER_HRU
+import by.homework.hlazarseni.tfgridexplorer.constants.DbConstants.CONVERTER_MRU
+import by.homework.hlazarseni.tfgridexplorer.constants.DbConstants.CONVERTER_SRU
+import by.homework.hlazarseni.tfgridexplorer.databinding.NodeDetailFragmentBinding
+import by.homework.hlazarseni.tfgridexplorer.lce.Lce
+import by.homework.hlazarseni.tfgridexplorer.services.GridProxyService
+import by.homework.hlazarseni.tfgridexplorer.viewModel.DetailViewModel
+import by.homework.hlazarseni.tfgridexplorer.util.TimeConverter.timeToString
+import coil.load
+import com.google.android.material.snackbar.Snackbar
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import org.koin.android.ext.android.inject
+
+
+class NodeDetailFragment : Fragment() {
+    private var _binding: NodeDetailFragmentBinding? = null
+    private val binding get() = requireNotNull(_binding)
+
+    private val args by navArgs<NodeDetailFragmentArgs>()
+
+   // private val viewModel by inject<DetailViewModel>()
```

Comment:
```suggestion
   private val viewModel by inject<DetailViewModel> { parametersOf(args.node) }
```
а сервис у тебя придет из графа

Reply:
шикарно)